Firestore/Example/Tests/Integration/API/FIRValidationTests.mm

```diff
@@ -228,7 +230,7 @@ - (void)testWritesWithInvalidTypesFail {
 }
 
 - (void)testWritesWithLargeNumbersFail {
-  NSNumber *num = @((unsigned long long)LONG_MAX + 1);
+  NSNumber *num = @(static_cast<uint64_t>(std::numeric_limits<int64_t>::max()) + 1);
```

Comment:
This works because of integer promotion.

The LHS has type `uint64_t` (aka `unsigned long long`) and the RHS has type `int`. `int` has a lower rank than than `long long` so the RHS s promoted to match the LHS. At that point the addition operates on uint64_t operands and the result a uint64_t whose value is one greater than LLONG_MAX.

See "Usual arithmetic conversions" in https://en.cppreference.com/w/c/language/conversion.

Reply:
Ah, I just missed that one is signed and the other is unsigned, so it was confusing.